inst/rmd/qc_report.rmd

```diff
@@ -103,32 +123,85 @@ if(has_filtered){
 ```
 
 ## Knee plot
-```{r}
+```{r, fig.alt="Knee plot of filtered and unfiltered droplets"}
 unfiltered_celldata <- data.frame(colData(unfiltered_sce)) %>%
   mutate(
-    rank = rank(- unfiltered_sce$sum), # using full spec for clarity 
+    rank = rank(- unfiltered_sce$sum, ties.method = "first"), # using full spec for clarity 
     filter_status = factor(ifelse(colnames(unfiltered_sce) %in% colnames(filtered_sce),
                                   "Passed", "Excluded"),
                            levels = c("Passed", "Excluded"))
-  )
+  ) %>%
+  filter(sum > 0) %>% # remove zeros for plotting
+  arrange(desc(rank))
 
 ggplot(unfiltered_celldata, aes(x = rank, y = sum, color = filter_status))+
-  geom_point(aes(shape = filter_status)) +
-  scale_y_log10() + 
-  scale_color_manual(values = c("navyblue", "grey")) + 
+  geom_point(aes(shape = filter_status),alpha = 0.2, size = 2) +
+  scale_x_log10(labels = scales::label_number()) +
+  scale_y_log10(labels = scales::label_number()) + 
+  scale_color_manual(values = c("darkgreen", "grey80")) + 
   scale_shape_manual(values = c(16, 1)) + # filled and unfilled circles
   labs(
     x = "Rank", 
-    y = "UMI count",
+    y = "Total UMI count",
     color = "Filter status",
     shape = "Filter status"
-  ) 
+  ) + 
+  guides(color = guide_legend(override.aes = list(alpha = 1))) + 
+  theme(plot.margin = margin(rep(20, 4)),
+        legend.position = c(0, 0),
+        legend.justification = c(0,0),
+        legend.background = element_rect(color = "grey20", size = 0.25),
+        legend.box.margin = margin(rep(5, 4)))
+```
+
+
+## Cell read metrics 
+
+```{r, fig.alt="Total UMI x genes expressed"}
+filtered_celldata <- data.frame(colData(filtered_sce)) 
+
+ggplot(filtered_celldata, 
+       aes (x = sum,
+            y = detected, 
+            color = subsets_mito_percent)) +
+  geom_point(alpha = 0.3) +
+  scale_color_viridis_c(limits = c(0, 100)) + 
+  labs(x = "Total UMI count",
+       y = "Number of genes detected",
+       color = "Percent reads\nmitochondrial") + 
+  theme(plot.margin = margin(rep(20, 4)),
+        legend.position = c(0, 1),
+        legend.justification = c(0,1),
+        legend.background = element_rect(color = "grey20", size = 0.25),
+        legend.box.margin = margin(rep(5, 4)))
+```
+
+## miQC model diagnostics
+
+```{r, fig.alt="miQC model diagnostics plot", results='asis', warning=FALSE}
+if(skip_miQC){
+  cat("miQC model not created, skipping miQC plot. Usually this is because mitochondrial gene data was not available.")
+}else{
+  # remove prob_compromised if it exists, as this will cause errors with plotModel
+  filtered_sce$prob_compromised <- NULL
+  miQC::plotModel(filtered_sce, model = metadata(filtered_sce)$miQC_model) + 
+    coord_cartesian(ylim = c(0,100)) +
+    labs(x = "Number of genes expressed",
```

Comment:
```suggestion
    labs(x = "Number of genes detected",
```
Just noticing that we have detected in the other plot so either we should change it here or change it to expressed in the other plot, I'm fine with either.